modules/linear_boltzmann_solvers/b_discrete_ordinates_solver/lbs_discrete_ordinates_solver.cc

```diff
@@ -480,7 +478,110 @@ DiscreteOrdinatesSolver::SetPrimarySTLvectorFromMultiGSPETScVecFrom(const std::v
     }
   } // for groupset id
 
-  VecRestoreArrayRead(x_src, &x_ref);
+  VecRestoreArrayRead(x, &x_ref);
+}
+
+void
+DiscreteOrdinatesSolver::ReorientAdjointSolution()
+{
+  for (const auto& groupset : groupsets_)
+  {
+    int gs = groupset.id_;
+
+    // Moment map for flux moments
+    const auto& moment_map = groupset.quadrature_->GetMomentToHarmonicsIndexMap();
+
+    // Angular flux info
+    auto& psi = psi_new_local_[gs];
+    const auto& uk_man = groupset.psi_uk_man_;
+
+    // Build reversed angle mapping
+    std::map<int, int> reversed_angle_map;
+    if (options_.save_angular_flux)
+    {
+      const auto& omegas = groupset.quadrature_->omegas_;
+      const auto num_gs_angles = omegas.size();
+
+      // Go through angles until all are paired
+      std::set<size_t> visited;
+      for (int m = 0; m < num_gs_angles; ++m)
+      {
+        // Skip if already encountered
+        if (visited.count(m) > 0) continue;
+
+        bool found = true;
+        for (int n = 0; n < num_gs_angles; ++n)
+        {
+          // Angles are opposite if their sum is zero
+          const auto sum = grid_ptr_->Attributes() & DIMENSION_1
+                             ? Vector3(0.0, 0.0, omegas[m].z + omegas[n].z)
+                             : omegas[m] + omegas[n];
+          const bool opposite = sum.NormSquare() < 1.0e-8;
+
+          // Add opposites to mapping
+          if (opposite)
+          {
+            found = true;
+            reversed_angle_map[m] = n;
+
+            visited.insert(m);
+            visited.insert(n);
+          }
+        } // for angle n
+
+        ChiLogicalErrorIf(not found,
+                          "Opposing angle for " + omegas[m].PrintStr() + " in groupset " +
+                            std::to_string(gs) + " not found.");
+
+      } // for angle m
+    }   // if saving angular flux
+
+    const auto num_gs_groups = groupset.groups_.size();
+    const auto gsg_i = groupset.groups_.front().id_;
+    const auto gsg_f = groupset.groups_.back().id_;
+
+    for (const auto& cell : grid_ptr_->local_cells)
+    {
+      const auto& transport_view = cell_transport_views_[cell.local_id_];
+      for (int i = 0; i < transport_view.NumNodes(); ++i)
+      {
+        // Reorient flux moments
+        //
+        // Because flux moments are integrated angular fluxes, the
+        // angular flux and spherical harmonics must be evaluated at
+        // opposite angles in the quadrature integration. Taking advantage
+        // of the even/odd nature of the spherical harmonics, i.e.
+        // Y_{\ell,m}(-\Omega) = (-1)^\ell Y_{\ell,m}(\Omega), the flux
+        // moments must be multiplied by (-1)^\ell.
+        for (int m = 0; m < num_moments_; ++m)
```

Comment:
In angular_quadrature_base.h, the HarmonicIndices struct has two entries: ell and m, following the standard math notation for spherical harmonics. 
Here, m is used has the running index for the mapping between the 2 indices {ell,m}(math) and the one index m (programming). Maybe m(programming) should be imom or n or something.

Reply:
Fair enough. I will change this.